drivers/digital-io/max22196/max22196.c

```diff
@@ -0,0 +1,557 @@
+				   MAX22196_CFG_MASK,
+				   no_os_field_prep(MAX22196_HITHR_MASK, hi_thr)
+				   | no_os_field_prep(MAX22196_CURR_MASK, curr));
+}
+
+/**
+ * @brief - MAX22196 filter set function
+ * @param desc - MAX22196 device descriptor.
+ * @param ch - Channel to which the filter values will be applied to.
+ * @param flt_en - Filter enable value(0 for filter disabled,
+ * 		   1 for filter enable)
+ * @param clr_filtr - Clear filter and set delay value to mid-scale.
+ * @param delay - Filter delay to be set.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22196_filter_set(struct max22196_desc *desc, uint32_t ch,
+			uint32_t flt_en, uint32_t clr_filtr,
+			enum max22196_delay delay)
+{
+	int ret;
+
+	if (ch > MAX22196_CHANNELS - 1)
+		return -EINVAL;
+
+	ret =  max22196_reg_update(desc, MAX22196_CFG_REG(ch),
+				   MAX22196_CFG_MASK,
+				   no_os_field_prep(MAX22196_DELAY_MASK, delay) |
+				   no_os_field_prep(MAX22196_FLTEN_MASK, flt_en));
+	if (ret)
+		return ret;
+
+	return max22196_reg_update(desc, MAX22196_GLOBALCFG_REG,
+				   MAX22196_FILTER_CLRFLT_MASK,
+				   no_os_field_prep(MAX22196_FILTER_CLRFLT_MASK,
+						   clr_filtr));
+}
+
+/**
+ * @brief - MAX22196 filter get function
+ * @param desc - MAX22196 device descriptor.
+ * @param ch - Channel from which the filter values are read.
+ * @param flt_en - Variable through which the fillter's state is read.
+ * @param clr_filtr - Clear the filter value and set to mid-scale delay.
+ * @param delay - Filter delay value to be read for the MAX22196's specified
+ * 		  channel.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22196_filter_get(struct max22196_desc *desc, uint32_t ch,
+			uint32_t *flt_en, uint32_t *clr_filtr,
+			enum max22196_delay *delay)
+{
+	int ret;
+	uint32_t del_val;
+
+	if (ch > MAX22196_CHANNELS - 1)
+		return -EINVAL;
+
+	ret = max22196_reg_read(desc, MAX22196_CFG_REG(ch), &del_val);
+	if (ret)
+		return ret;
+
+	*delay = no_os_field_get(MAX22196_DELAY_MASK, del_val);
+	*flt_en = no_os_field_get(MAX22196_FLTEN_MASK, del_val);
+
+	ret = max22196_reg_read(desc, MAX22196_GLOBALCFG_REG, &del_val);
+
+	*clr_filtr = no_os_field_get(MAX22196_FILTER_CLRFLT_MASK, del_val);
+
+	return 0;
+}
+
+/**
+ * @brief Set fault mask bits in the fault mask registers.
+ * @param desc - MAX22196 device descriptor.
+ * @param fault_mask - Requested fault to set mask to.
+ * @param enabled - Mask state to be set.
+ * @return - 0 in case of succes, negative error code otherwise.
+*/
+int max22196_fault_mask_set(struct max22196_desc *desc,
+			    enum max22196_fault_mask fault_mask,
+			    bool enabled)
+{
+	switch (fault_mask) {
+	case MAX22196_GLOBAL_REFDISHTCFG:
+		return max22196_reg_update(desc, MAX22196_GLOBALCFG_REG,
+					   MAX22196_FAULT_MASK(fault_mask),
+					   no_os_field_prep(MAX22196_FAULT_MASK(fault_mask),
+							   enabled));
+	case MAX22196_FAULT1_VMLOW ... MAX22196_FAULT1_FAULT2:
```

Comment:
the logic from max22190 should be followed here as well,  regarding fault2 enable bit in fault1 register